lib/Multihash.ts

```diff
@@ -76,4 +77,30 @@ export default class Multihash {
     const multihashEncodedString = Encoder.encode(multihashBuffer);
     return multihashEncodedString;
   }
+
+  /**
+   * Checks if the given hash is a multihash computed using the configured hashing algorithm.
+   */
+  public static validateHashComputedUsingSupportedHashAlgorithm (
+    encodedMultihash: string,
+    inputContextForErrorLogging: string
+  ) {
+    let multihash;
+    try {
+      const multihashBuffer = Encoder.decodeAsBuffer(encodedMultihash);
+      multihash = multihashes.decode(multihashBuffer);
+    } catch {
+      throw new IonError(ErrorCode.MultihashStringNotAMultihash, `Given ${inputContextForErrorLogging} string '${encodedMultihash}' is not a multihash.`);
```

Comment:
Nit: technically the string could still be a multihash, just not encoded in the write encoding scheme (e.g. base58) on line 90. So perhaps generalize the error message, or break into two `try` block, or move the check out by accepting just buffer.

Reply:
Changed along with the comment above